docs/source/llm-eval/llm-eval.rst

```diff
@@ -0,0 +1,470 @@
+            "- Score 1: Language is casual but generally respectful and avoids strong informality or slang. Acceptable in "
+            "some informal professional settings."
+            "- Score 2: Language is overall formal but still have casual words/phrases. Borderline for professional contexts."
+            "- Score 3: Language is balanced and avoids extreme informality or formality. Suitable for most professional contexts. "
+            "- Score 4: Language is noticeably formal, respectful, and avoids casual elements. Appropriate for formal "
+            "business or academic settings. "
+        ),
+        examples=[professionalism_example_score_2, professionalism_example_score_4],
+        model="openai:/gpt-3.5-turbo-16k",
+        parameters={"temperature": 0.0},
+        aggregations=["mean", "variance"],
+        greater_is_better=True,
+    )
+
+Best Practice
+~~~~~~~~~~~~~~
+
+Under the hood of evaluting with LLM as the judge is prompt engineering, and how to find the best prompt is still under research. 
+Here are some tips for setting required fields of custom SaaS LLM evaluation metrics:
+
+* TODO(sunish): please share your learnings here.
+* TODO(prithvi): please share your learnings here.
+* TODO(ann): please share your learnings here.
+
+
+Create Per-row LLM Evluation Metrics (Cateogory 2)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This is very similar to creating a custom traditional metrics, with the exception of returning a `EvaluationResult` instance.
+Basically you need to:
+
+1. Implement a ``eval_fn`` to define your scoring logic, it must take in 3 args ``predictions``, ``targets`` and ``metrics``.
+   ``eval_fn`` must return a :py:func:`mlflow.metrics.MetricValue` instance.
+2. Pass ``eval_fn`` and other arguments to ``mlflow.metricsmake_metric`` API to create the metric. 
+
+The following code creates a dummy per-row metric called ``"over_10_chars"``: if the model output is greater than 10, 
+the score is 1 otherwise 0.
+
+.. code-block:: python
+
+    def eval_fn(predictions, targets, metrics):
+        scores = []
+        for i in range(len(predictions)):
+            if len(predictions[i]) > 10:
+                scores.append(1)
+            else:
+                scores.append(0)
+        return MetricValue(
+            scores=scores,
+            aggregate_results=standard_aggregations(scores),
+        )
+
+
+    # Create an EvaluationMetric object.
+    passing_code_metric = make_metric(
+        eval_fn=eval_fn, greater_is_better=False, name="over_10_chars"
+    )
+
+
+Prepare Your LLM for Evaluating
+=====================================
+
+In order to evaluate your LLM with ``mlflow.evaluate()``, your LLM has to be one of the following type:
+
+1. A :py:func:`mlflow.pyfunc.PyFuncModel` instance or a URI pointing to a logged `mlflow.pyfunc.PyFuncModel` model. In
+   general we call that MLflow model.
+2. A python function that takes in string inputs and outputs a single string. 
```

Comment:
Yea I saw that requirement, but feel confused about what it means. for example the following works as a custom callable for evaluate:
```
def openai_qa(inputs):
    answers = []
    system_prompt = "Please answer the following question in formal language."
    for index, row in inputs.iterrows():
        completion = openai.ChatCompletion.create(
            model="gpt-3.5-turbo",
            messages=[
                {"role": "system", "content": system_prompt},
                {"role": "user", "content": "{row}"},
            ],
        )
        answers.append(completion.choices[0].message.content)

    return answers
```
And it just returns a python list?

Reply:
yea, that works because its list -> list, which is a valid predict fn. see https://mlflow.org/docs/latest/python_api/mlflow.pyfunc.html#mlflow.pyfunc.PyFuncModel.predict

Reply:
got it, let me add it